src/Client/Campaign.php

```diff
@@ -17,6 +17,11 @@ public function getById(string $id): array
     {
         try {
             $response = $this->getHttpClient()->get("{$this->getSetting('campaign', 'baseUri')}/$id");
+
+            if (!json_decode($response->getBody()->getContents(), true)['ready']) {
+                // uncaught exception
+                throw new CampaignNotReadyException('Campaign not ready');
```

Comment:
I'm not sure we gain much from having a distinct exception class here – we want behaviour to be as close to the existing 'not found' case as possible I think, including the sandbox-specific cases we already covered.

If we must have a new Exception class it should extend the existing one so that inheritance leads all the existing catches to continue working as they do now. But given there is a distinct exception message, I'm not convinced we need a separate class at all.